src/main/java/org/generationcp/breeding/manager/customcomponent/DeletePrefixCacheDialog.java

```diff
@@ -0,0 +1,354 @@
+		this.prefixesTable.setColumnWidth(DeletePrefixCacheDialog.PREFIXES_PROPERTY_ID, 350);
+		this.prefixesTable.setColumnAlignment(DeletePrefixCacheDialog.PREFIXES_PROPERTY_ID, Table.ALIGN_CENTER);
+		this.prefixesTable.setColumnAlignment(DeletePrefixCacheDialog.REMOVE_PROPERTY_ID, Table.ALIGN_CENTER);
+	}
+
+	private void addPrefix() {
+		final String prefix = this.prefixTextField.getValue().toString().trim();
+		if(!StringUtil.isEmpty(prefix)) {
+			if(!this.prefixesTable.getVisibleItemIds().contains(prefix.toUpperCase())) {
+				final Item newItem = this.prefixesTable.getContainerDataSource().addItem(prefix.toUpperCase());
+				newItem.getItemProperty(DeletePrefixCacheDialog.PREFIXES_PROPERTY_ID).setValue(prefix);
+
+				final Button removeButton = new Button("REMOVE");
+				removeButton.addListener(new RemovePrefixButtonListener(this));
+				removeButton.setData(prefix.toUpperCase() );
+				newItem.getItemProperty(DeletePrefixCacheDialog.REMOVE_PROPERTY_ID).setValue(removeButton);
+				this.prefixesTable.requestRepaint();
+				this.deletePrefixesButton.setEnabled(true);
+			} else {
+				MessageNotifier
+					.showWarning(this.source.getWindow(), this.messageSource.getMessage(Message.WARNING),
+						this.messageSource.getMessage(Message.WARNING_PREFIX_EXISTING_IN_TABLE));
+			}
+		}
+
+		this.prefixTextField.setValue("");
+	}
+
+	private void deleteFromTable(final String itemId) {
+		this.prefixesTable.getContainerDataSource().removeItem(itemId);
+		this.prefixesTable.requestRepaint();
+		if(CollectionUtils.isEmpty(this.prefixesTable.getVisibleItemIds())) {
+			this.deletePrefixesButton.setEnabled(false);
+		}
+	}
+
+	void deletePrefixes() {
+		final List<String> prefixes = new ArrayList<>((Collection<? extends String>) this.prefixesTable.getVisibleItemIds());
+		final List<String> names = this.germplasmDataManager.getNamesByGidsAndPrefixes(this.deletedGIDs, prefixes);
+		if(!CollectionUtils.isEmpty(names)) {
+			final List<KeySequenceRegister> keySequenceRegistersOfDeletedGermplasm = this.germplasmDataManager.getKeySequenceRegistersByPrefixes(prefixes);
+			if(CollectionUtils.isEmpty(keySequenceRegistersOfDeletedGermplasm)) {
+				MessageNotifier
+					.showError(this.source.getWindow(), this.messageSource.getMessage(Message.ERROR),
+						this.messageSource.getMessage(Message.NO_EXISTING_NAME_WITH_PREFIX));
+			} else {
+				this.deleteKeyRegisters(names, keySequenceRegistersOfDeletedGermplasm);
+			}
+		} else {
+			MessageNotifier
+				.showError(this.source.getWindow(), this.messageSource.getMessage(Message.ERROR),
+					this.messageSource.getMessage(Message.NO_EXISTING_NAME_WITH_PREFIX));
+		}
+		this.cancelButton.click();
+	}
+
+	void deleteKeyRegisters(final List<String> names, final List<KeySequenceRegister> keySequenceRegistersOfDeletedGermplasm) {
+		final Set<String> prefixesToBeDeleted = new HashSet<>();
+
```

Comment:
For performance improvement, we should only compile the pattern once per prefix (do it outside for loop)